cmake/extensions.cmake

```diff
@@ -1219,7 +1219,7 @@ function(generate_unique_target_name_from_filename filename target_name)
   string(REPLACE "." "_" x ${basename})
   string(REPLACE "@" "_" x ${x})
 
-  string(RANDOM LENGTH 8 random_chars)
+  string(MD5 unique_chars ${filename})
```

Comment:
Is filename relative or absolute?

Can we get into trouble of we invoke 'generate_unique_target_name_from_filename' twice on two different generated files with the same relative address?

If not, then this looks good :)

Reply:
Thanks for questioning this. I've done some research and experimentation. I think this collision issue:
- might be possible in theory,
- is extremely unlikely in practice,
- if ever, will trigger a very clear error message like this one:
```
CMake Error at zephyr/cmake/extensions.cmake:512 (add_custom_target):
  add_custom_target cannot create target
  "gen_lets_encrypt_x3_der_inc_7ea3a619643a01cb9f7b7914adc3ad38" because
  another target with the same name already exists.  The existing target is a
  custom target created in source directory
  "zephyr/samples/net/sockets/big_http_download".
  See documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  src2/CMakeLists.txt:7 (generate_inc_file_for_target)
````
To trigger the above error message I hacked `samples/net/sockets/big_http_download/` and invoked `generate_inc_file_for_target()` with the same `include/generated//lets_encrypt_x3.der.inc` relative argument twice. This is really not how CMakeLists.txt files are normally written because CMake generally hates/discourages/deprecates relative paths https://cmake.org/cmake/help/v3.3/variable/CMAKE_USE_RELATIVE_PATHS.html (gone in 3.4)

I scanned the current code base. There are currently eight places where this function is used: 

1. `cfb.cmake: generate_cfb_font_for_target()`
```
samples/display/cfb_custom_font/
```
2. `extensions.cmake:  generate_inc_file_for_target()`
```
samples/net/sockets/big_http_download/
samples/net/sockets/dumb_http_server/
samples/net/sockets/echo_client/
samples/net/sockets/echo_server/
samples/net/sockets/http_get/
soc/arm/nxp_lpc/lpc54xxx/
tests/application_development/gen_inc_file/
```
I traced these invocations and they all use absolute paths because they all rely after a (scary...) number of indirections on `MAKE_CURRENT_BINARY_DIR`. CMake's love of absolute paths makes sure `CMAKE_CURRENT_BINARY_DIR` is one https://cmake.org/cmake/help/v3.13/variable/CMAKE_CURRENT_BINARY_DIR.html

Thoughts?

Reply:
Thank you for the thorough analysis. I agree with your findings and conclusion that this is safe.